include/seqan3/range/view/kmer_hash.hpp

```diff
@@ -0,0 +1,102 @@
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \brief Provides seqan3::view::kmer_hash.
+ */
+
+#pragma once
+
+#include <range/v3/view/sliding.hpp>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/range/view/detail.hpp>
+#include <seqan3/std/view/all.hpp>
+#include <seqan3/std/view/transform.hpp>
+
+namespace seqan3::detail
+{
+// ============================================================================
+//  kmer_hash_fn (adaptor definition)
+// ============================================================================
+
+//!\brief View adaptor definition for view::kmer_hash.
+class kmer_hash_fn : public pipable_adaptor_base<kmer_hash_fn>
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = pipable_adaptor_base<kmer_hash_fn>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief       Call the view's constructor with the underlying view as argument.
+     * \returns     A range of converted elements.
+     */
+    template <std::ranges::View urng_t>
+        requires semi_alphabet_concept<value_type_t<urng_t>>
+    static auto impl(urng_t urange, size_t const k)
+    {
+        return urange | ranges::view::sliding(k) | view::transform(
+        [] (auto const in)
+        {
+            std::hash<decltype(in)> h{};
+            return h(in);
+        });;
+    }
+
+    /*!\brief       Call the view's constructor with the underlying range wrapped in seqan3::view::all as argument.
+     * \returns     A range of converted elements.
+     */
+    template <std::ranges::ViewableRange urng_t>
+        requires semi_alphabet_concept<value_type_t<urng_t>>
+    static auto impl(urng_t && urange, size_t const k)
+    {
+        return impl(view::all(std::forward<urng_t>(urange)), k);
+    }
```

Comment:
You don't need these two overload, you only need one with ViewableRange as constraint!